### ISSUE TITLE
Uptick in API errors (for products.delete and products.insert)

### ISSUE THREAD

**Issue by findingsimple** [woocommerce/google-listings-and-ads#1181]
### Describe the bug:

We've started seeing an uptick in errors since the end of 2021 in products.delete and products.insert.

As of this week we're now also seeing errors with productstatuses.get and accounts.claimwebsite

The Google team have done some digging and believe it to be a _timing_ or retry issue - it looks to them like duplicate requests are being fired off.

### Expected behavior:

Reduced errors / no duplicate requests.

## Additional Information

* Error rates can be monitored from within our MC account UI
* A list of MC IDs and a slide deck outlining some screenshots is available within our call notes (Search for "Uptick in API errors")

**Comment by mikkamp** [comment]
Based on the investigation we did for these errors the largest amount of errors was `notFound`. Which occurs when the extension tries to interact with products that are no longer present in the Merchant Center. One of the reasons this could happen is because a user manually deletes products from the Merchant Center.

The issues #1193 and #1194 have been created to target the specific reasons why the list of MC ID's might not be up to date.

There was also a high failure rate for the claimwebsite request, which turned out to be from one account which wasn't created by our MCA. The Google team will followup on this and there is nothing to do from our side.

Closing this issue in favour of the followup issues.
